app/hooks/useJsonSchemaDiff.ts

```diff
@@ -0,0 +1,92 @@
+import diff from 'deep-diff';
+
+const addObjectToDiffMap = (path, lhs, rhs, idDiffMap) => {
+  if (typeof lhs !== 'object' && typeof rhs !== 'object') {
+    idDiffMap[path] = {lhs: lhs ?? null, rhs: rhs ?? null};
+    return;
+  }
+  if (typeof lhs === 'object' || typeof rhs === 'object') {
+    const keys = new Set<string>();
+
+    if (lhs) Object.keys(lhs).forEach((k) => keys.add(k));
+    if (rhs) Object.keys(rhs).forEach((k) => keys.add(k));
+
+    keys.forEach((k) =>
+      addObjectToDiffMap(`${path}_${k}`, lhs?.[k], rhs?.[k], idDiffMap)
+    );
+  }
+};
+
+/**
+ *
+ * @param formData the form data to display when not diffing
+ * @param showDiff whether or not the diff should be performed
+ * @param formIdPrefix
+ * @param lhs
+ * @param rhs
+ * @returns An object with the formData to display, and the idDiffMap, containing the field-level diffs
+ */
+const useJsonSchemaDiff = (
+  formData: any,
+  showDiff: boolean,
+  formIdPrefix: string,
+  lhs?: any,
+  rhs?: any
+) => {
+  if (!showDiff) return {formData};
+  const idDiffMap: Record<string, {lhs: any; rhs: any}> = {};
+  const differences = diff(lhs, rhs);
+  let newFormData = Array.isArray(rhs) ? [...rhs] : {...rhs};
+
+  if (differences) {
+    // Populate the diffPathArray and diffArray
+    differences.forEach((difference) => {
+      if (difference.kind === 'A') {
+        // Array differences
+        const arrayElementPath = [
+          formIdPrefix,
+          ...(difference.path ?? []),
+          difference.index
+        ].join('_');
+        if (difference.item.kind === 'N') {
+          // Added an element to the array
+          Object.keys(difference.item.rhs).forEach((key) => {
+            idDiffMap[`${arrayElementPath}_${key}`] = {
+              lhs: null,
+              rhs: difference.item.rhs[key]
+            };
+          });
+        } else if (difference.item.kind === 'D') {
+          const deletedItem = {};
+          // Deleted an element from the array
+          Object.keys(difference.item.lhs).forEach((key) => {
+            deletedItem[key] = null;
```

Comment:
`deletedItem` doesn't seem to be used anywhere?

Reply:
Indeed, that was from a previous implementation that didn't work. This injects `difference.item.lhs` below instead, so that the conditional schemas render properly